gpudrive/visualize/core.py

```diff
@@ -235,79 +257,45 @@ def plot_simulator_state(
                 alpha=1.0,
                 line_width_scale=line_width_scale,
                 marker_size_scale=marker_scale,
-                extended_goals=extended_goals
+                extended_goals=extended_goals,
+                policy_mask=policy_mask
             )
 
             if agent_positions is not None:
-                # First calculate the maximum valid trajectory length across all agents for this env_idx
-                max_valid_length = 0
                 for agent_idx in range(agent_positions.shape[1]):
                     if controlled_live[agent_idx]:
                         trajectory = agent_positions[env_idx, agent_idx, :time_step, :]
-                        valid_mask = ((trajectory[:, 0] != 0) & (trajectory[:, 1] != 0) &
```

Comment:
I think you didn't rebase this correctly over main -- why are you deleting this code?